improver/cli/hail_size.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# (C) British Crown copyright. The Met Office.
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""CLI to generate hail size."""
+from improver import cli
+
+
+@cli.clizefy
+@cli.with_output
+def process(
+    *cubes: cli.inputcube, ccl_cubes: cli.inputcubelist, model_id_attr: str = None
+):
+    """Module to calculate the size of hail stones from the
+    cloud condensation level temperature and pressure, temperature
+    on pressure levels data and relative humidity on pressure levels.
+
+    Args:
+        cubes (iris.cube.CubeList or list of iris.cube.Cube):
+            containing:
+                temperature (iris.cube.Cube):
+                    Cube of temperature on pressure_levels
+                relative_humidity (iris.cube.Cube)
+                    Cube of relative_humidity on pressure levels
+        ccl_cubes (iris.cube.CubeList)
+            Cube list containing a cube of air temperature at ccl
+            and a cube of air pressure at ccl
+
+        model_id_attr (str):
+            Name of the attribute used to identify the source model for blending.
+
+    Returns:
+        iris.cube.Cube:
+            Cube of size_of_hail (mm).
```

Comment:
That makes sense as a convention but to predict the size of hail stones in meters seems very wrong given the scale (I really hope hail stones don't get to 1 meter in diameter). It makes more sense to me to keep it as mm but if you feel strongly about this I'm happy to be overruled.

Reply:
Yeah, not really a feeling issue, more of a standards issue. @MoseleyS may be able to suggest this is all internal and thus doesn't need to conform, but otherwise it should be metres.

Reply:
I've check with Stephen and he said he'd be happy with it in meters so I've updated it all to meters